### PR TITLE
webgui: moved events controller to events service

### PR DESCRIPTION
**Pull request by ralder**

### CODE REVIEW

File: gui/scripts/syncthing/core/services/events.js [hunk 1]
```diff
@@ -17,39 +17,47 @@ angular.module('syncthing.core')
...
+            }, 1000, false);
+        }
 
-        $http.get(urlbase + '/events?limit=1')
-            .success(successFn)
-            .error(errorFn);
-    });
+        angular.extend(self, {
+            ONLINE: 'UIOnline',
+            OFFLINE: 'UIOffline',
```
Comment (calmh) [inline gui/scripts/syncthing/core/services/events.js]: Why these and not the other constants while anyway doing this?

Comment (ralder) [inline gui/scripts/syncthing/core/services/events.js]: it seems you are right))

**Comment by calmh** [comment]
:+1:

**Comment by snnd** [comment]
@ralder @calmh is there currenty a general refactoring going on? i see lots of people talking about a new or custom webui, too. regarding to the current state of webui source quality and angularjs version, it probably might be better idea to rewrite the webui from scratch. that's what i already thought some months ago when i started refactoring the first time.

what do both of you think about that idea?

**Comment by calmh** [comment]
I don't think it's *that* bad, but you're certainly welcome to propose a new UI.

**Comment by snnd** [comment]
The hardest part will be to extract functionality from controller into separate services while not spoiling the �loading sequence�. This is what I ended up some months ago. On the other hand it�ll be a great experience refactoring an application of this style.

I started out with directive extraction this time. More details on this come in a separate isse/pr.

Dennis John Wilson

T (0)176 6226 3882
www.abakia.de
dennis.wilson@abakia.de

**Pull request merged**